vyper/builtin_functions/functions.py

```diff
@@ -851,6 +855,9 @@ def build_LLL(self, expr, args, kwargs, context):
         elif sub.location == "storage":
             lengetter = LLLnode.from_list(["sload", "_sub"], typ=BaseType("int128"))
             elementgetter = _storage_element_getter
+        elif sub.location == "calldata":
+            lengetter = LLLnode.from_list(["calldatasize"], typ="uint256")
+            elementgetter = _calldata_element_getter
         # TODO: unclosed if/elif clause.  Undefined behavior if `sub.location`
```

Comment:
While you're in here could you get rid of the unclosed else clause? Throw an exception instead of undefined behavior.